cmd/gke-certificates-controller/app/options.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package app implements a server that runs a stand-alone version of the
+// certificates controller.
+package app
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/spf13/pflag"
+)
+
+// GKECertificatesController is the main context object for the package.
+type GKECertificatesController struct {
+	Kubeconfig                    string
+	Master                        string
+	ClusterSigningCertFile        string
+	ClusterSigningKeyFile         string
+	ClusterSigningGKEKubeconfig   string
+	ClusterSigningGKERetryBackoff metav1.Duration
+	ApproveAllKubeletCSRsForGroup string
+	UseServiceAccountCredentials  bool
+	ServiceAccountKeyFile         string
+	RootCAFile                    string
+}
+
+// Create a new instance of a GKECertificatesController with default parameters.
+func NewGKECertificatesController() *GKECertificatesController {
+	s := &GKECertificatesController{
+		ClusterSigningCertFile:        "/etc/kubernetes/ca/ca.pem",
+		ClusterSigningKeyFile:         "/etc/kubernetes/ca/ca.key",
```

Comment:
Definitely fair. This also falls under the "cloning kube-controller-manager's flags" umbrella. With the GKE-specific rebranding, I believe I can tear these out entirely since we'll always want to call the GKE endpoint instead of having the CA private keys on the masters.